proto/emergency_light.proto

```diff
@@ -0,0 +1,98 @@
+syntax = "proto3";
+
+package smartcore.bos;
+
+option go_package = "github.com/vanti-dev/sc-bos/pkg/gen";
+
+import "google/protobuf/duration.proto";
+import "google/protobuf/field_mask.proto";
+import "google/protobuf/timestamp.proto";
+
+// EmergencyLightApi provides methods to start and stop tests on emergency lights, and to retrieve test results.
+// The API is designed to be implemented by emergency lights.
+service EmergencyLightApi {
+  // Attempt to start a function test
+  rpc StartFunctionTest(StartEmergencyTestRequest) returns (StartEmergencyTestResponse);
+  // Attempt to start a duration test
+  rpc StartDurationTest(StartEmergencyTestRequest) returns (StartEmergencyTestResponse);
+  // Stop any test that is in progress. If no tests are running this has no effect.
+  // When a running test is stopped before completion, its EmergencyTestResult.Result MUST be updated to TEST_RESULT_UNSPECIFIED
+  rpc StopEmergencyTest(StopEmergencyTestsRequest) returns (StopEmergencyTestsResponse);
+  // Get the set of results of the most recent tests performed on the emergency light.
+  rpc GetTestResultSet(GetTestResultSetRequest) returns (TestResultSet);
+  rpc PullTestResultSets(PullTestResultRequest) returns (stream PullTestResultsResponse);
+}
+
+// TestResultSet contains the results of the most recent tests performed on the emergency light.
+message TestResultSet {
+  EmergencyTestResult function_test = 1;
+  EmergencyTestResult duration_test = 2;
+}
+
+message EmergencyTestResult {
+  enum Result {
+    TEST_RESULT_UNSPECIFIED = 0;
+    TEST_PASSED = 1;
+    CIRCUIT_FAILURE = 2;
+    BATTERY_DURATION_FAILURE = 3;
+    BATTERY_FAILURE = 4;
+    LAMP_FAILURE = 5;
+    TEST_FAILED = 6;
+    LIGHT_FAULTY = 7;
+    COMMUNICATION_FAILURE = 8;
+    OTHER_FAULT = 9;
+    TEST_RESULT_PENDING = 10;
```

Comment:
This is a minor thing, but I find it useful to know that enum values over a certain ordinal are classed together. For example

```go
if t.Result.Number() > 2 {
  return err
}
```

It means you can check for errors and still react correctly even if new error states are added in the future. We can't do that with this enum ordering.

Consider putting PENDING near the top, maybe at `1` and move all the others down?